src/tribler/gui/widgets/downloadspage.py

```diff
@@ -459,12 +494,16 @@ def on_export_download(self, checked):
             self.dialog.show()
 
     def on_export_download_dialog_done(self, action):
+        def on_finish(result: Tuple):
+            data, _ = result
+            self.on_export_download_request_done(filename, data)
+
         selected_item = self.selected_items[:1]
         if action == 0 and selected_item:
             filename = self.dialog.dialog_widget.dialog_input.text()
-            TriblerFileDownloadRequest(
+            FileDownloadRequest(
                 f"downloads/{selected_item[0].download_info['infohash']}/torrent",
-                lambda data: self.on_export_download_request_done(filename, data),
+                on_finish,
             )
 
```

Comment:
Forgot to call `request_manager.add(request)`. This is an example that a new API can be a bit error-prone, as the call of `request_manager.add` can be easily forgotten.

For that reason, I suggest adding lightweight methods to `RequestManager`: `get`, `post`, `put`, etc., that create a corresponding request object and immediately add it to the request manager. With these methods, it would be impossible to forget to call `request_manager.add`. As a side benefit, it makes the codebase more concise again.